war/src/main/js/add-item.js

```diff
@@ -0,0 +1,369 @@
+// Initialize all modules by requiring them. Also makes sure they get bundled (see gulpfile.js).
+var $ = require('jquery-detached').getJQuery();
+
+var getItems = function(){
+  var d = $.Deferred();
+  $.get('categories?depth=3').done(
+      function(data){
+        d.resolve(data);
+      }
+  );
+  return d.promise();
+}; 
+
+var jRoot = $('head').attr('data-rooturl');
+
+$.when(getItems()).done(function(data){
+  $(function() {
+
+    // The main panel content is hidden by default via an
+    // inline style. We're ready to remove that now.
+    $('#create-item-panel').removeAttr('style');
+
+    //////////////////////////////
+    // helpful reference DOM
+
+    var defaultMinToShow = 2;
+    var defaultLooseItems = 'jenkins.category.uncategorized';
+    var $root = $('#main-panel');
+    var $form = $root.find('form[name="createItem"]').addClass('jenkins-config new-view');
+    var $newView = $('<div class="new-view" />')
+      .attr('name','createItem')
+      .attr('action','createItem')
+      .prependTo($form);
+    var $tabs = $('<div class="jenkins-config-widgets" />').appendTo($newView);
+    var $categories = $('<div class="categories" />').appendTo($newView);
+    var $subBtn = $('#bottom-sticker .yui-submit-button');
+    var sectionsToShow = [];
+
+    ////////////////////////////////
+    // submit button click
+    
+    function makeButtonWrapper(){
+      var $p = window.$; // jshint ignore:line
```

Comment:
@gusreiber Is it not obvious that these lines warrant some comments for people trying to maintain this code? I was looking at them for a few mins before I realised that you're hooking into prototype (I think). I'm somewhat familiar with the landscape here, so I doubt other readers would have a clue.

Next question ... why do we need to use prototype for this?